### ISSUE TITLE
Storybook stuck loading

### ISSUE THREAD

**Issue by thany** [storybookjs/storybook#17329]
**Describe the bug**
In an attempt to update from 6.4.9 to 6.4.14, after building and opening the browser tab, Storybook initially loads into its loading state, but it's stuck there. It never completes loading its UI, like this:

![image](https://user-images.githubusercontent.com/152227/150963102-a546fc54-3108-4b3c-b04f-c5c985c41835.png)

There are no build errors/warnings, no console errors and no meaningful warnings, and no network errors.

Only thin I can see is `/progress` which is presumably a part of Storybook, doesn't finish loading.

I've also upgraded `@storybook/preset-create-react-app` from 3.2.0 to 4.0.0, and in the changelog, all I can see is support for CRA 5, which is all fine - I should assume CRA v4 is not removed, is it? Because that one hadn't been upgraded. However, downgrading this package back to 3.2.0 works around the problem: Storybook 6.4.14 now loads correctly.

**To Reproduce**
Follow the above description.

**System**
Please paste the results of `npx sb@next info` here.

```
  System:
    OS: Windows 10 10.0.19043
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
  Binaries:
    Node: 16.13.2 - C:\Program Files\nodejs\node.EXE
    npm: 8.3.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 97.0.4692.99
    Edge: Spartan (44.19041.1266.0), Chromium (97.0.1072.62)
  npmPackages:
    @storybook/addon-a11y: 6.4.14 => 6.4.14
    @storybook/addon-actions: 6.4.14 => 6.4.14
    @storybook/addon-controls: 6.4.14 => 6.4.14
    @storybook/addon-docs: 6.4.14 => 6.4.14
    @storybook/addons: 6.4.14 => 6.4.14
    @storybook/preset-create-react-app: ^3.2.0 => 3.2.0
    @storybook/react: 6.4.14 => 6.4.14
    @storybook/theming: 6.4.14 => 6.4.14
```

(How does this still manage to miss the one browser I'm actually using??)

**Additional context**
Other packages noteworthy:
* Craco 6.2.0 - using Craco means we have no direct control over whichever CRA version it uses.
* Webpack 4.46.0 - Updating to 5.x gave us a myriad of weird problems, so we decided to stick with v4, which should be fine because that one isn't EOL yet.

I'm in the process of updating (or trying to, honestly) our dependencies to their latest, and I would like small chunks of updates at a time. Updating too many packages in one go makes it difficult to diagnose problems. So yeah, I could also update Craco, but I don't want to just yet, because it's not only Storybook depending on it.

**Comment by nvonbenken** [comment]
Having the same problem with webpack 5 on the latest prerelease.

```
  System:
    OS: Linux 5.10 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
  Binaries:
    Node: 16.13.2 - /tmp/fnm_multishells/400_1643292508391/bin/node
    npm: 8.1.2 - /tmp/fnm_multishells/400_1643292508391/bin/npm
  npmPackages:
    @storybook/addon-actions: ^6.5.0-alpha.23 => 6.5.0-alpha.23
    @storybook/addon-docs: ^6.5.0-alpha.23 => 6.5.0-alpha.23
    @storybook/addon-links: ^6.5.0-alpha.23 => 6.5.0-alpha.23
    @storybook/addon-storyshots: ^6.5.0-alpha.23 => 6.5.0-alpha.23
    @storybook/addons: ^6.5.0-alpha.23 => 6.5.0-alpha.23
    @storybook/builder-webpack5: ^6.5.0-alpha.23 => 6.5.0-alpha.23
    @storybook/manager-webpack5: ^6.5.0-alpha.23 => 6.5.0-alpha.23
    @storybook/preset-create-react-app: ^4.0.0 => 4.0.0
    @storybook/react: ^6.5.0-alpha.23 => 6.5.0-alpha.23
    ```

**Comment by shilman** [comment]
There are probably two different problems going on here.

@thany I think your use case of using webpack4 with CRA5 is not supported by the preset.

@nvonbenken I think you are hitting a bug starting in 6.5.0-alpha.23. you can downgrade to 6.5.0-alpha.22 or wait for a fix which should be forthcoming.

**Comment by shilman** [comment]
ZOMG!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.5.0-alpha.26 containing PR #17361 that references this issue. Upgrade today to the `@next` NPM tag to try it out!

```
npx sb upgrade --prerelease
```

Closing this issue. Please re-open if you think there's still more to do.

**Comment by johannespein** [comment]
I have the same issue. Tried different versions, nothing worked.

```System:
    OS: Linux 5.15 Ubuntu 22.04 LTS 22.04 LTS (Jammy Jellyfish)
    CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
  Binaries:
    Node: 14.19.0 - ~/.nvm/versions/node/v14.19.0/bin/node
    Yarn: 1.22.17 - /home/linuxbrew/.linuxbrew/bin/yarn
    npm: 6.14.16 - ~/.nvm/versions/node/v14.19.0/bin/npm
  Browsers:
    Chrome: 101.0.4951.54
    Firefox: 100.0
  npmPackages:
    @storybook/addon-actions: ^6.4.22 => 6.4.22 
    @storybook/addon-essentials: ^6.4.22 => 6.4.22 
    @storybook/addon-interactions: ^6.4.22 => 6.4.22 
    @storybook/addon-links: ^6.4.22 => 6.4.22 
    @storybook/node-logger: ^6.4.22 => 6.4.22 
    @storybook/preset-create-react-app: ^3.2.0 => 3.2.0 
    @storybook/react: ^6.4.22 => 6.4.22 
    @storybook/testing-library: ^0.0.11 => 0.0.11 ```

**Comment by RishikeshVedpathak** [comment]
The issue still persists even after upgrading storybook to 6.5.3

**Comment by trainoasis** [comment]
Have the same issue after upgrading to 6.5.4. Any tips appreciated

**Comment by shilman** [comment]
@trainoasis I believe that loading error happens when you use `@storybook/preset-create-react-app@4` with `react-scripts@4`. If this matches your setup, either upgrade to `react-scripts@5` or downgrade to `@storybook/preset-create-react-app@3`. More discussion #18221

**Comment by trainoasis** [comment]
> @trainoasis I believe that loading error happens when you use `@storybook/preset-create-react-app@4` with `react-scripts@4`. If this matches your setup, either upgrade to `react-scripts@5` or downgrade to `@storybook/preset-create-react-app@3`. More discussion #18221

it does, and have tried that, thanks - in general I want to upgrade to `react-scripts@5` but unfortunately that brings up many other issues so it seems it'll have to wait (even though there's many security vulnerabilities with the old version).

**Comment by b-a-merritt** [comment]
Be careful that this behavior is not being caused by the component hierarchy having two consecutive `/`. I spent some time thinking I needed to downgrade react-scripts when it was that I had mistakenly written something like: 
```TypeScript
export default {
   title: "Atoms//Button
} as Meta
```

**Comment by MuhammadSawalhy** [comment]
There should be some error message from Storybook to help us identify what the problem is and what causes it.

For now and until Storybook take some actions, we can use `git bisect` to identify what commit caused this problem.

**Comment by aommm** [comment]
> There should be some error message from Storybook to help us identify what the problem is and what causes it.

Agree, it would be great with some error messages in the "infinite spinner" case.
I'm trying to set up storybook for my project now, but after running into the infinite spinner I'm not sure how to proceed. (I probably have some problem with my Webpack config, but it's hard to tell)

**Comment by NourIM** [comment]
I'm having same issue and having deadline to deliver the storybook and I'm sooooo blocked :(

**Comment by velidan** [comment]
The same problem

**Comment by velidan** [comment]
@NourIM try to check the dev tools console. I had the same problem but then I checked the console and found that some import was undefined

**Comment by NourIM** [comment]
@velidan :(
<img width="1283" alt="Screenshot 2022-11-09 at 11 26 47" src="https://user-images.githubusercontent.com/14943545/200819506-2566676a-3fad-4c13-ad9a-2e79c8e45405.png">
<img width="824" alt="Screenshot 2022-11-09 at 11 26 29" src="https://user-images.githubusercontent.com/14943545/200819496-bebbeb60-ebcd-4b7a-8c37-0287a29463c5.png">

**Comment by johannespn** [comment]
This should not be closed :) @shilman

**Comment by shilman** [comment]
@johannespn What's left to do here?

**Comment by zemfr** [comment]
In my case all requests were rewritten to index.html. I added rewrite rule fore iframe.html

**Comment by christineurban** [comment]
Not sure if this will help anyone but after a few hours of debugging I finally got rid of the stuck loading by fixing this little mistake:

```
import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { Heading as HeadingComponent } from "../components/heading";

export default {
  title: "Heading",
  component: HeadingComponent,
} as ComponentMeta<typeof HeadingComponent>;


// bad - VSCode doesn't highlight this as error for me so I couldn't see the issue
export const Heading: ComponentStory<typeof HeadingComponent> = () => (
  <Heading <--- this
    size="xxl"
    level="h1"
    children="This is an xxl heading"
  />
);

// good
export const Heading: ComponentStory<typeof HeadingComponent> = () => (
  <HeadingComponent
    size="xxl"
    level="h1"
    children="This is an xxl heading"
  />
);

```

**Comment by glassdimlygr** [comment]
This happened to me on FF when I had too many storybook tabs open. I don't know why, but opening in Chrome / closing all my FF tabs with SB in them made the whitescreening go away.

**Comment by arimendelow** [comment]
agreed, same issue, please reopen. in my case it's caused by passing a specific prop into the component, i have no clue why it's causing this issue. zero logging whatsoever

**Comment by kajurek** [comment]
I'm facing the same issue using: 
"@nrwl/storybook": "15.3.0"
"@storybook/addon-essentials": "~6.5.9",
"@storybook/builder-vite": "0.2.5",
"@storybook/core-server": "~6.5.9",
"@storybook/react": "~6.5.9",


please reopen and fix the issue

**Comment by B1RO** [comment]
For me this happens in a monorepo. When my story imports one of my monorepo packages/a component that uses that package, the storybook ends up infinitely loading. There is no error.

**Comment by gayancharith** [comment]
also facing the same issue.
```
    "@storybook/react": "6.5.16",
    "@storybook/addon-essentials": "6.5.16",
    "@storybook/addon-viewport": "6.5.16",
    "@storybook/preset-create-react-app": "4.1.2",
    "storybook-addon-next-router": "^4.0.2",
```

`ERR! Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/username/node_modules/chalk/source/index.js from /Users/username/node_modules/@storybook/react/node_modules/@storybook/node-logger/dist/cjs/index.js not supported.`

`ERR! Instead change the require of /Users/username/node_modules/chalk/source/index.js in /Users/username/node_modules/@storybook/react/node_modules/@storybook/node-logger/dist/cjs/index.js to a dynamic import() which is available in all CommonJS modules.`

**Comment by clevertree** [comment]
this issue is hitting people at random and is likely caused by corporate sabotage paid for by competition like figma. We have to wait for the developers to be not paid off to fix this one

**Comment by NathanVss** [comment]
We are facing this exact same issue too. 😕

**Comment by blcoyote** [comment]
Tagging along on this.
![image](https://user-images.githubusercontent.com/30732216/226621013-5ce8a7f6-9170-4d7a-b00d-78f5934dec9d.png)
/progress is 2.4 minutes loading on a NX single app monorepo  "@nrwl/storybook": "^15.8.6",

**Comment by tonalnik** [comment]
In my case, I used some context provider, that was `export default abstract class SomeContextService`. I change it to 
```
abstract class SomeContextService{ ... }
export default SomeContextService
```
and it starts working for some reason

**Comment by chuanqisun** [comment]
Same issue.
- Monorepo
- Ubuntu 22.04
- React + TypeScript + Vite

![image](https://github.com/storybookjs/storybook/assets/1895289/82698511-8ffa-43dd-a61a-7d33d74b032e)


![image](https://github.com/storybookjs/storybook/assets/1895289/100fa49f-ad81-42c1-9bee-1ea25d27d14f)
![image](https://github.com/storybookjs/storybook/assets/1895289/4b02de73-df8d-4c15-8f52-477152f8859e)

With barebone `main.ts`

```ts
import type { StorybookConfig } from "@storybook/react-vite";

const config: StorybookConfig = {
	stories: ["../src/**/button.stories.@(js|jsx|ts|tsx)"],
	addons: ["@storybook/addon-essentials"],
	framework: {
		name: "@storybook/react-vite",
		options: {},
	},
};
export default config;
```

And barebone `preview.ts`
```ts
import type { Preview } from "@storybook/react";

const preview: Preview = {
	parameters: {
		actions: { argTypesRegex: "^on[A-Z].*" },
		controls: {
			matchers: {
				color: /(background|color)$/i,
				date: /Date$/,
			},
		},
	},
};

export default preview;
```

**Comment by lockykeaney** [comment]
Same thing has happened to me. It was working fine 10 minutes ago, and then I tried to add [storybook-preset-inline-svg](https://www.npmjs.com/package/storybook-preset-inline-svg) and got the infinite load.

Removed the dependencies and re-ran `yarn install` but the issue persists.

No real indication of the issue, but this error might be related:
```
Uncaught (in promise) Error: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was received
```

Version 6.5.16, and I can't upgrade because the project is React 17 and react-native-web which isn't supported in the new version, I've tried.

**Comment by lockykeaney** [comment]
@clevertree Yeah, done the whole nuke and everything, but no change.

I might have to completely remove my install of storybook and give it another shot.

**Comment by nurit-levi-hs** [comment]
can somene refer me to a fix please?

**Comment by basickarl** [comment]
Still getting these issues where Storybook refuses to load without any information.

**Comment by Mrowa96** [comment]
Same here, it's really frustrating

**Comment by alshdotme** [comment]
Same here on 8.5. Only in Safari 18.2 though. Works in other browsers.

**Comment by mak-koda** [comment]
This is still an issue, Storybook just refuses to finish loading without any information or errors.

**Comment by sopost-kirt** [comment]
For me, running `rm -rf ./node_modules && npm install` fixed it

**Comment by PaRoxUs** [comment]
I found the reason why our components were stuck in an "infinite spinner" case.

It was the config of moduel-federation/vite in vite.config plugins that broke Storybook.

Because we don't need to use module federation when using Storybook, I solved the issue by conditionally including the config.

**Comment by justindomingue** [comment]
Running `storybook build` revealed a build error, which when addressed, fixed the infinite loading issue.

**Comment by aymericzip** [comment]
I got a similar issue by deploying my storybook 

stack : vite / react / docker 

In the browser console, I saw 
`It looks like you're visiting the Storybook development server on another hostname than localhost: 'storybook.intlayer.org', but you haven't configured the necessary security features to support this.
Please re-run your Storybook development server with the '--host storybook.intlayer.org' flag, or manually configure your Vite allowedHosts configuration with viteFinal.`

I changed the start command to
`    "start": "pnpm storybook -p 6006 --no-open --host 0.0.0.0",`

then got another 

```
info => Starting manager..
info => Starting preview..
info Using tsconfig paths for react-docgen
[intlayer]  Dictionaries built
;2;241;97;97mError: listen EADDRNOTAVAIL: address not available 173.249.57.38:6006
    at Server.setupListenHandle [as _listen2] (node:net:1918:21)
    at listenInCluster (node:net:1997:12)
    at GetAddrInfoReqWrap.callback (node:net:2206:7)
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:134:8)

WARN Broken build, fix the error above.
WARN You may need to refresh the browser.

(node:29) [DEP0190] DeprecationWarning: Passing args to a child process with shell option true can lead to security vulnerabilities, as the arguments are not escaped, only concatenated.
(Use `node --trace-deprecation ...` to show where the warning was created)
```

So I had to set `allowedHosts` as

```ts
import type { StorybookConfig } from '@storybook/react-vite';
import { defineConfig, mergeConfig } from 'vite';
import { intlayerPlugin } from 'vite-intlayer';

const config: StorybookConfig = {
  // ...
  async viteFinal(config, { configType }) {
    config.server = {
      ...config.server,
      host: true,
      allowedHosts: ['storybook.intlayer.org', 'localhost', '127.0.0.1'],
    };

    return config 
};

export default config;
```
